pyvista/plotting/renderer.py

```diff
@@ -293,9 +294,52 @@ def _charts(self):
         # lazy instantiation here to avoid creating the charts object unless needed.
         if self.__charts is None:
             self.__charts = Charts(self)
-            self.AddObserver("StartEvent", partial(try_callback, self._render_event))
+            self.add_render_event(self._before_render_event, "before")
         return self.__charts
 
+    def add_render_event(self, callback, event="before"):
```

Comment:
I added these methods at an earlier stage in this PR for a workaround that is no longer needed at this point. I think it's still ok to keep these in though?

Reply:
I think this method should be `interaction_event`, not `add_render_event` as the VTK event types are `StartEvent` and `EndEvent`. Further, this obfuscates the `add_on_render_callback()` I added in #3385.

Reply:
Oof, I think I may need to refactor `add_on_render_callback()` to be on the `Renderer`...